src/Analyzer/Passes/FunctionToSubcolumnsPass.cpp

```diff
@@ -297,6 +297,12 @@ class FunctionToSubcolumnsVisitorFirstPass : public InDepthQueryTreeVisitorWithC
             return;
         }
 
+        if (const auto * /*cross_join_node*/ _ = node->as<CrossJoinNode>())
+        {
+            can_wrap_result_columns_with_nullable |= getContext()->getSettingsRef()[Setting::join_use_nulls];
```

Comment:
Actually, we do not convert to nullable in the case of a cross join. However, it should also check the join kind above as well (i.e., do not convert for inner). It seems we just won’t apply certain optimizations if we set this flag to conservative, correct?

Reply:
I think so.
So far, I have disabled it for a cross-join the same way it was.
Let's check join kind in the separate PR.